### PR TITLE
sagemath: update to 10.3.

### PR DESCRIPTION
**Pull request by tornaria**
PR for tracking sagemath beta/rc until 10.3 is released.

#### Testing the changes
- I tested the changes in this PR: **briefly**

### CODE REVIEW

File: srcpkgs/sagemath/template [hunk 1]
```diff
@@ -26,14 +26,14 @@ depends="eclib-devel fflas-ffpack flintlib-devel gcc-fortran gd-devel
...
+checkdepends="$depends pythran python3-Sphinx"
 short_desc="Open source mathematics software"
 maintainer="Gonzalo Tornaría <tornaria@cmat.edu.uy>"
 license="GPL-2.0-or-later"
 homepage="https://www.sagemath.org/"
 changelog="https://github.com/sagemath/sage/releases"
 distfiles="https://github.com/sagemath/sage/archive/refs/tags/$version.tar.gz"
-checksum=e7125f13495e1068edab73735aca7f9b2c655688096e9d109e628c023e76411f
+checksum=3f72b8d13d8bb472daaeabff4feba98e392028328e20012980da9c457dc49fb0
 nocross="due to ntl (eclib), fflas-ffpack, givaro, linbox, sympow, maxima"
```
Comment (dkwo) [inline srcpkgs/sagemath/template:37]: Can you add `singular` here, just so we don't forget?
`nocross="due to ntl (eclib, singular), fflas-ffpack, givaro, linbox, sympow, maxima"`

Comment (tornaria) [reply srcpkgs/sagemath/template:37]: But singular cross builds just fine, doesn't it? Is the cross-built singular not suitable to build sagemath?

Comment (dkwo) [reply srcpkgs/sagemath/template:37]: indeed, the cross-built singular (i.e. built without ntl) gives a lot of segfaults in sage doctests.

Comment (dkwo) [reply srcpkgs/sagemath/template:37]: yes, i agree.
sounds good.

**Comment by tornaria** [comment]
@ahesford testing here cython 3.0.7 to compile and check sagemath

**Comment by ahesford** [comment]
Thanks for checking Cython; updated in 0dea09d39debbab95eeb01110423a619e370498c.

**Comment by tornaria** [comment]
@dkwo sagemath 10.3 is close to being released. This PR is up-to-date with rc1, using flintlib 3.1 (it's just simpler to update both at the same time -- singular needs patching as well).

Also I changed (slightly) the package, first to use pep517 build (which works fine) and also moved the internal scripts from `/usr/lib/sagemath/10.3/bin` directly into `/usr/libexec/sagemath`. Both changes should be transparent to the user.

If you want to test (esp. in aarch64) let me know of any trouble. We have a small window to contribute fixes upstream if something pops up.

**Comment by tornaria** [comment]
I added the patch to givaro; revbump to be safe since the missing include is in a header that we ship in givaro-devel.

**Comment by tornaria** [comment]
I'm getting weird "error" in CI (no log at all), I don't know what is going on...

**Comment by tornaria** [comment]
It turns out that the `flintlib-3.1.0.tar.gz` tarball had changed (to fix two issues I reported, see https://github.com/flintlib/flint/issues/1809) hence the template was failing at fetching it. It's updated now.

**Comment by tornaria** [comment]
> I'm getting weird "error" in CI (no log at all), I don't know what is going on...
It seems the logs are now XHRed from a `*.windows.net` address instead of `*.github.com`, which umatrix was blocking. This confused me as it seemed the CI itself was completely broken.

**Comment by dkwo** [comment]
It all builds and passes checks on aarch64 now, thanks!

**Comment by tornaria** [comment]
@dkwo final release of sagemath 10.3, if you want to review and give one more try before we merge.

Notes
- gap: I'm holding back 4.13.0 since there is no PR available on sagemath yet (https://github.com/sagemath/sage/issues/37616)
- singular: updated to 4.3.2p16 for flint 3.1 support, needs a small https://github.com/sagemath/sage/pull/37492; I added myself as co-maintainer
- switched to pep517 build, but still from a github tarball; to build from pypi sdist we are missing https://github.com/sagemath/sage/pull/37287

**Comment by dkwo** [comment]
This builds and checks fine on aarch64 :)
time to merge it? @leahneukirchen

**Comment by leahneukirchen** [comment]
Then mark as ready.

**Comment by tornaria** [comment]
Ready to merge! Thanks @dkwo @leahneukirchen for your support.

**Pull request merged**